### PR TITLE
ToDo List

### PR DESCRIPTION
**Pull request by Shur111k**

### CODE REVIEW

File: src/App.css [hunk 1]
```diff
@@ -1,38 +1,92 @@
...
+
+* {
+  font-family: 'Poppins', sans-serif;
+  margin: 0;
+  padding: 0;
+  box-sizing: border-box;
+}
+
+body {
+  background: #8758ff;
```
Comment (Vovchyk-Mahuran) [inline src/App.css]: colors must be in variables. Please remember it and do it always

File: src/App.css [hunk 2]
```diff
@@ -1,38 +1,92 @@
...
+  color: #fff;
+  margin-bottom: 0.5rem;
+  font-size: 1.75rem;
 }
 
-@media (prefers-reduced-motion: no-preference) {
-  .App-logo {
-    animation: App-logo-spin infinite 20s linear;
-  }
+.TodoWrapper {
```
Comment (Vovchyk-Mahuran) [inline src/App.css]: why different format of classnames? Use BEM

File: src/components/EditToolForm.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import React, {useState} from 'react'
+
+export const EditTodoForm = ({editTodo, task}) => {
+    const [value , setValue] = useState(task.task)
+    const handleSubmit = e =>{
```
Comment (Vovchyk-Mahuran) [inline src/components/EditToolForm.js]: enter before

File: src/components/EditToolForm.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+        e.preventDefault();
+
+        editTodo(value, task.id);
+
+        setValue("")
+    }
+  return (
+    <form className='TodoForm' onSubmit={handleSubmit}>
+        <input type='text' className='todo-input' value={value} placeholder='Update Task' onChange={(e) => setValue(e.target.value)}/>
```
Comment (Vovchyk-Mahuran) [inline src/components/EditToolForm.js]: create a separate handler for this, and avoid anonymous functions usage iniside of markup. You have to use it as onChange={handleChangeValue}

File: src/components/Todo.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import React from 'react'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { faPenToSquare } from '@fortawesome/free-solid-svg-icons'
+import { faTrash } from '@fortawesome/free-solid-svg-icons'
+
+export const Todo = ({task, toggleComplete, deleteTodo, editTodo}) => {
+  return (
+    <div className='Todo'>
+        <p onClick={() => toggleComplete(task.id)} className={`${task.completed ? 'completed':""}`}>{task.task}</p>
```
Comment (Vovchyk-Mahuran) [inline src/components/Todo.js]: move to separate handler

Comment (Vovchyk-Mahuran) [reply src/components/Todo.js]: Also install 'classnames' library for preventing usage of ternary operator

File: src/components/TodoForm.js [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+
+        setValue("")
+    }
+  return (
+    <form className='TodoForm' onSubmit={handleSubmit}>
+        <input type='text' className='todo-input' value={value} placeholder='What is the task to do?' onChange={(e) => setValue(e.target.value)}/>
+        <button type='submit 'className='todo-btn'>Add Task</button>
+    </form>
+  )
+}
```
Comment (Vovchyk-Mahuran) [inline src/components/TodoForm.js]: Hmm, I have seen this before.

File: src/components/TodoWrapper.js [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+import { Todo } from './Todo';
+import { EditTodoForm } from './EditToolForm';
+uuidv4();
+
+export const TodoWrapper = () => {
+    const [todos, setTodos] = useState([])
+
+    const addTodo = todo => {
+        setTodos([...todos, {id: uuidv4(), task: todo,
+        completed: false, isEditing: false}])
```
Comment (Vovchyk-Mahuran) [inline src/components/TodoWrapper.js]: read about state which depends on previous state value in react

File: src/components/TodoWrapper.js [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+            todo.isEditing ? (
+                <EditTodoForm editTodo={editTask} task={todo}/>
+            ) : (
+                <Todo task={todo} key={index}
+                toggleComplete={toggleComplete} 
+                deleteTodo={deleteTodo}
+                editTodo={editTodo}/>
+            )
+        ))}
+    </div>
```
Comment (Vovchyk-Mahuran) [inline src/components/TodoWrapper.js]: 1. Read about indexes as a key

2.Its better to move the list to separate component

File: src/components/Form/index.js [hunk 9]
```diff
@@ -0,0 +1,34 @@
+import React, { useState } from 'react';
+import styles from './style.module.css';
+
+export const TodoForm = ({ initialValue = '', handleSubmit, placeholder, buttonText}) => {
+
+  console.log(initialValue)
```
Comment (Vovchyk-Mahuran) [inline src/components/Form/index.js]: remove logs

File: src/components/Todo/index.js [hunk 10]
```diff
@@ -0,0 +1,54 @@
+import React, { useState } from 'react';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faPenToSquare, faTrash } from '@fortawesome/free-solid-svg-icons';
+import { TodoForm } from '../Form'
+import classNames from 'classnames';
+import styles from './style.module.css';
+
+export const Todo = ({ task, toggleComplete, deleteTodo, editTodo}) => {
+
```
Comment (Vovchyk-Mahuran) [inline src/components/Todo/index.js]: remove this enter

File: src/components/Todo/index.js [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+        onClick={handleToggleComplete} 
+        className={classNames({
+          [styles.completed]: task.completed,
+          [styles.incompleted]: !task.completed
+        })}
+      >
+        {task.task}
+      </p>
+      <div>
+          <>
```
Comment (Vovchyk-Mahuran) [inline src/components/Todo/index.js]: why fragments?

Comment (Shur111k) [reply src/components/Todo/index.js]: can remove, because they are in <dіv>'?

Comment (Vovchyk-Mahuran) [reply src/components/Todo/index.js]: yes

**Review by Vovchyk-Mahuran — APPROVED** [review]

**Pull request merged**